src/metabase/query_processor/expand.clj

```diff
@@ -393,21 +393,47 @@
 (s/defn ^:ql ^:always-validate expressions
   "Top-level clause. Add additional calculated fields to a query."
   {:added "0.17.0"}
-  [query, m :- {s/Keyword Expression}]
+  [query, m :- {s/Keyword i/AnyExpression}]
   (assoc query :expressions m))
 
-(s/defn ^:private ^:always-validate expression-fn :- Expression
+(s/defn ^:private ^:always-validate expression-fn :- ArithmeticExpression
   [k :- s/Keyword, & args]
-  (i/map->Expression {:operator k, :args (vec (for [arg args]
-                                                (if (number? arg)
-                                                  (float arg) ; convert args to floats so things like 5 / 10 -> 0.5 instead of 0
-                                                  arg)))}))
+  (i/map->ArithmeticExpression {:operator k, :args (vec (for [arg args]
+                                                          (if (number? arg)
+                                                            (float arg) ; convert args to floats so things like 5 / 10 -> 0.5 instead of 0
+                                                            arg)))}))
 
 (def ^:ql ^{:arglists '([rvalue1 rvalue2 & more]), :added "0.17.0"} + "Arithmetic addition function."       (partial expression-fn :+))
 (def ^:ql ^{:arglists '([rvalue1 rvalue2 & more]), :added "0.17.0"} - "Arithmetic subtraction function."    (partial expression-fn :-))
 (def ^:ql ^{:arglists '([rvalue1 rvalue2 & more]), :added "0.17.0"} * "Arithmetic multiplication function." (partial expression-fn :*))
 (def ^:ql ^{:arglists '([rvalue1 rvalue2 & more]), :added "0.17.0"} / "Arithmetic division function."       (partial expression-fn :/))
 
+(s/defn ^:ql sql-expression
+  "Expression representing arbitrary SQL."
+  [sql-expr :- su/NonBlankString]
+  (i/map->SQLExpression {:expr sql-expr}))
+
+(s/defn ^:ql remap-expression
```

Comment:
@tlrobinson suggested `remap-inline` for this and `remap-lookup` instead of `fk-remap-expression`. I'm a :+1: on better names for this.